DREAM3DReviewFilters/GenerateMaskFromSimpleShapes.cpp

```diff
@@ -0,0 +1,669 @@
+
+        if(inBounds)
+        {
+          m_Mask[i] = true;
+        }
+
+        k++;
+      }
+    }
+  }
+}
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void GenerateMaskFromSimpleShapes::createVertexMask()
+{
+  size_t totalNumCenters = m_CentersPtr.lock()->getNumberOfTuples();
+  size_t totalNumElementsDest = m_MaskPtr.lock()->getNumberOfTuples();
+  DataContainer::Pointer dc = getDataContainerArray()->getDataContainer(getMaskArrayPath().getDataContainerName());
+  VertexGeom::Pointer vertex = dc->getGeometryAs<VertexGeom>();
+
+  bool inBounds = false;
+
+  float currentx = 0;
+  float currenty = 0;
+  float currentz = 0;
+
+  float* vertices = vertex->getVertexPointer(0);
+  size_t k = 0;
+
+  if(m_MaskShape == 0)
+  {
+    for(size_t i = 0; i < totalNumElementsDest; i++)
+    {
+      inBounds = false;
+      k = 0;
+      currentx = vertices[3 * i];
+      currenty = vertices[3 * i + 1];
+      currentz = vertices[3 * i + 2];
+
+      while(!inBounds && k < totalNumCenters)
+      {
+        inBounds = IsPointInEllipsoidBounds(m_Centers[3 * k], m_Centers[3 * k + 1], m_Centers[3 * k + 2], m_AxisLengths[3 * k], m_AxisLengths[3 * k + 1], m_AxisLengths[3 * k + 2], currentx, currenty,
+                                            currentz);
+
+        if(inBounds)
+        {
+          m_Mask[i] = true;
+        }
+
+        k++;
+      }
+    }
+  }
+
+  if(m_MaskShape == 1)
+  {
+    for(size_t i = 0; i < totalNumElementsDest; i++)
+    {
+      inBounds = false;
+      k = 0;
+      currentx = vertices[3 * i];
+      currenty = vertices[3 * i + 1];
+      currentz = vertices[3 * i + 2];
+
+      while(!inBounds && k < totalNumCenters)
+      {
+        inBounds = IsPointInBoxBounds(m_Centers[3 * k], m_Centers[3 * k + 1], m_Centers[3 * k + 2], m_BoxDims[3 * k], m_BoxDims[3 * k + 1], m_BoxDims[3 * k + 2], currentx, currenty, currentz);
+
+        if(inBounds)
+        {
+          m_Mask[i] = true;
+        }
+
+        k++;
+      }
+    }
+  }
+
+  if(m_MaskShape == 2)
+  {
+    for(size_t i = 0; i < totalNumElementsDest; i++)
+    {
+      inBounds = false;
+      k = 0;
+      currentx = vertices[3 * i];
+      currenty = vertices[3 * i + 1];
+      currentz = vertices[3 * i + 2];
+
+      while(!inBounds && k < totalNumCenters)
+      {
+        inBounds = IsPointInCylinderBounds(m_Centers[3 * k], m_Centers[3 * k + 1], m_Centers[3 * k + 2], m_CylinderRad[k], m_CylinderHeight[k], currentx, currenty, currentz);
+
+        if(inBounds)
+        {
+          m_Mask[i] = true;
+        }
+
+        k++;
+      }
+    }
+  }
+}
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void GenerateMaskFromSimpleShapes::createEdgeMask()
+{
+}
```

Comment:
Unimplemented?

Reply:
That can be removed for now. I haven't gotten to that implementation. There should be something in data check (line 107) that doesn't allow edge geometries right now, so you should never end up here.

Reply:
removed this in latest commit